### ISSUE TITLE
Rewrite build system in Literally Anything Other Than Bash

### ISSUE THREAD

**Issue by clo4** [clo4/datapacks#20]
Not literally anything, but most *would* be better. Leaning towards Python, but using Deno does make setting up a watcher very easy and it supports this kind of use-case well. I've already used Deno as a build system in [apple_device_identifiers](https://github.com/clo4/apple_device_identifiers)

**Comment by clo4** [comment]
More thinking. There shouldn't be one major build script, instead each project should have its own that can support its own features, and there can be a sort of orchestration script that executes all of them in parallel in the root directory.

**Comment by clo4** [comment]
So here's my current idea. There will be an executable file in each datapack's directory called `build.fish`, using fish for its sane shell scripting syntax. This is the logic responsible for zipping the contents and creating the datapack. Each build script will create a zip file called `datapack.zip` in the current directory. As part of the Nix flake builder, it will then rename this datapack according to the name and version specified.

Using the builder should look something like this:
```nix
{ buildDataPack }:
buildDataPack {
  pname = "afk";
  version = "1.2";
  minMinecraftVersion = "1.13.x";
  maxMinecraftVersion = "1.21.0";
  src = ./.;
}
```

And the build script might look like this:
```fish
set all (ls)
mkdir -p build
cp -R $all build
cd build

for dir in **/function
  cp 
end

zip -r datapack.zip LICENSE pack.png pack.mcmeta version.json data
```
obviously there could be more complicated logic there, but that's a starting point and for most should be all that's necessary.

**Comment by clo4** [comment]
... and that's basically what I did. New build system landed in b4cb016, except it doesn't use a build script, the compat logic is handled by the `preBuild` phase. The build phase is just zipping the contents of the `zip` list, and the install phase is moving it to the correct location with the correct name. The output contents will be the same as they would have been with the old build script.
